brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -83,7 +83,10 @@ public SpanId startNewSpan(String component, String operation) {
     }
 
     private SpanId getNewSpanId() {
-        Span parentSpan = spanAndEndpoint().state().getCurrentServerSpan().getSpan();
+        Span parentSpan = spanAndEndpoint().state().getCurrentLocalSpan();
```

Comment:
I don't want to break back-compat here even if folks having properly finished their spans. I'm curious if folks intentionally do not finish spans or if it is simply due to mistakes. If the former, is that as an intentional filter of sorts? If the latter, I'd probably want to know as that means you're silently losing spans (though you'd see gaps in your traces).

Curious about your thoughts at implementation approach here. I could do an instanceof check on the state to see if its `InheritableServerClientAndLocalSpanState` and only then inherit the local parent span, but as you mentioned that feels really brittle. Slightly better might be an interface that state implementations implement to control their parent span inheritence, something along the lines of:

```
public interface InheritableState {
    /**
     * @return the current parent span to inherit, or null if no parent span should be inherited
     */
    @Nullable
    Span getCurrentParentSpan();
}
```